### ISSUE TITLE
Improved handling or UX when using a bad connection

### ISSUE THREAD

**Issue by atduarte** [mingo-app/mingo#596]
Using Mingo with a bad internet connection is pretty impossible. I understand that it can't work miracles with the SSH connection, but the UX could be significantly improved. It's unclear what's the state of the connection, how to reconnect, and doesn't recover easily even when connected to a new stable internet connection.

**Comment by tothradoslav** [comment]
Yeah, i agree with you. Can you elaborate on this a bit, how do you imagine this to be ideally implemented? Thank you !

**Comment by tothradoslav** [comment]
There have been many updates on how the underlying system works. Closing issue. Please let me know if there are any specific issues, create a new issue. Thank you!
